R/utils.R

```diff
@@ -1,81 +1,144 @@
+        pattern = paste0("^", file_name, "\\.sas7bdat$"),
+        x = candidates,
+        ignore.case = TRUE,
+        value = TRUE
+      )
+
+      # Prefer SAS file if it exists, otherwise use RDS
+      if (isTRUE(prefer_sas)) {
+        if (length(sas_match) > 0) {
+          return(file.path(dir_path, sas_match[1]))
+        } else if (length(rds_match) > 0) {
+          return(file.path(dir_path, rds_match[1]))
+        } else {
+          stop(dir_path, " does not contain SAS or RDS file: ", file_name)
+        }
+      } else if (isFALSE(prefer_sas)) {
+        if (length(rds_match) > 0) {
+          return(file.path(dir_path, rds_match[1]))
+        } else if (length(sas_match) > 0) {
+          return(file.path(dir_path, sas_match[1]))
+        } else {
+          stop(dir_path, " does not contain RDS or SAS file: ", file_name)
+        }
       }
-
-      index <- match(uppercase_file_name, uppercase_candidates)
-      if (!is.na(index)) {
-        file_name_to_load <- candidates[[index]]
-        break
+    } else {
+      # If an extension is provided, use the exact file name
+      if (file.exists(file_path)) {
+        return(file_path)
+      } else {
+        stop(dir_path, " does not contain: ", file_name)
       }
     }
-
-    if (is.null(file_name_to_load)) {
-      stop(paste("create_data_list(): No RDS or SAS files found for", file_path, x))
-    }
-
-    output <- read_file(file_path, file_name_to_load)
-
-    return(output)
   })
 
-  names(data_list) <- file_names
-
-  return(data_list)
+  # Normalize all file paths
+  return(normalizePath(unlist(file_paths)))
 }
 
 
-#' Reads RDS/SAS file and metadatas from first 6 items from file.info() its file path
-#' @param file_path a path to a file
-#' @param file_name name of a file
-#' @return a data object with an extra attribute of metadata
-read_file <- function(file_path, file_name) {
-  ext <- tools::file_ext(file_name)
 
-  if (!(toupper(ext) %in% c("RDS", "SAS7BDAT"))) {
-    stop("Usage error: read_file: file_name: file must either be RDS or SAS7BDAT.")
-  }
+#' Load Data Files
+#'
+#' This function reads data from multiple file paths and returns a list of data frames.
+#' It supports reading RDS and SAS7BDAT files.
+#'
+#' @param file_paths [character(1+)] A vector of file paths to read.
+#'
+#' @return [list] A named list of data frames, where each name is the basename of the corresponding file path.
+#'
+#' @examples
+#' path <- system.file("examples", "iris.sas7bdat", package = "haven")
+#' data_list <- load_data_files(file_paths = path)
+#' str(data_list)
+#'
+#' @export
+load_data_files <- function(file_paths) {
+  # Validate input parameters
+  checkmate::assert_character(file_paths, min.len = 1)
+  checkmate::assert_file_exists(file_paths)
+
+  # Read each file and store in a list
+  data_list <- lapply(file_paths, function(file_path) {
+    # Get file extension
+    extension <- tools::file_ext(file_path)
+
+    # Read file based on its extension
```

Comment:
I don't get the rewrite from `toupper` to `tolower`, the change in the error message, the moving of code around and the obvious comments. They only open the doors for a bug to creep in and make code review a longer process for no discernible benefit.

Reply:
The old `toupper` is only used to do a case-insensitive check of the file extension. It's a non-user-facing implementation detail. Changing the logic is unnecessary.